superset/sql_validators/trino.py

```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Type
+
+from superset.sql_validators.presto_db import PrestoDBSQLValidator
```

Comment:
Unfortunately I haven't had a chance to test this (and I'm on a task unrelated to Trino, so it'll be a while before I have a chance to test), but I have some real concerns over re-using the PrestoDB Validator for this. First off, Trino was forked from Presto several years ago now and the projects continue to diverge with each release. But more specifically, I was looking at that PrestoDBSQLValidator class and it does a bunch of error handling that assumes that the underlying library is PyHive. PyHive is no longer the recommended way to connect Superset to Trino and it hasn't been for quite some time. Imo, this implementation should take that into consideration and it should be independent from PrestoDB / PyHive.

Reply:
We've been using this in our fork for the past year or so I believe, just to give us some sort of validation for trino queries. It's true I've occasionally seen some errors logged e.g. a missing `poll` attribute on `Cursor` which suggests they're not fully compatible here, and have to fudge the error type does flag the issue. We've found that it's better than nothing, but perhaps not reliable enough to be a useful contribution.

We're likely to go the route of highly customised validation errors and warnings on the validator if we do write a more comprehensive version, but if we take inspiration from what the presto and postgres validators are looking for maybe we can build that in such a way to be able to contribute something useful here and add anything more custom on top.

That said it sounds like trino isn't a big priority in this project  at the moment, and the SQL validators aren't a big focus either, given there have only ever been two of them. Hopefully we can contribute something more useful here once we've iterated it a bit, though.